otter/models/cass.py

```diff
@@ -1682,6 +1694,54 @@ def health_check(self):
         return d
 
 
+@implementer(IScalingGroupServersCache)
+class CassScalingGroupServersCache(object):
+    """
+    Collection of cache of scaling group servers
+    """
+
+    def __init__(self, tenant_id, group_id):
+        self.tenant_id = tenant_id
+        self.group_id = group_id
+        self.table = "servers_cache"
+        self.params = {"tenant_id": tenant_id, "group_id": group_id}
+
+    @do
+    def get_servers(self):
+        query = ("SELECT server_blob, last_update FROM {cf} "
+                 "WHERE tenant_id=:tenant_id AND group_id=:group_id "
+                 "ORDER BY last_update DESC;")
+        rows = yield cql_eff(query.format(cf=self.table), self.params)
+        if len(rows) == 0:
+            yield do_return(([], None))
+        last_update = rows[0]['last_update']
+        rows = take_while(lambda r: r['last_update'] == last_update, rows)
+        yield do_return(([json.loads(r['server_blob']) for r in rows],
+                         last_update))
+
+    def insert_servers(self, last_update, servers, clear_others):
+        query = ("INSERT INTO {cf} (tenant_id, group_id, last_update, "
+                 "server_id, server_blob "
+                 "VALUES(:tenant_id, :group_id, :last_update, :server_id{i}, "
```

Comment:
I think the error I was getting was that there were no queries (because there are no servers), but the empty query was still being executed with params.  You asked if there was a case where there wouldn't be a cache - and I totally missed this obvious one.  :(  Apologies.  Which means that we get the last 30 days always if there are no servers on the group.  :|  Do we need to provide a sentinel value to stop it?